photon-core/src/main/java/org/photonvision/vision/camera/USBCameras/GenericUSBCameraSettables.java

```diff
@@ -0,0 +1,306 @@
+/*
+ * Copyright (C) Photon Vision.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package org.photonvision.vision.camera.USBCameras;
+
+import edu.wpi.first.cscore.UsbCamera;
+import edu.wpi.first.cscore.VideoException;
+import edu.wpi.first.cscore.VideoMode;
+import edu.wpi.first.cscore.VideoProperty;
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.util.PixelFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.photonvision.common.configuration.CameraConfiguration;
+import org.photonvision.common.configuration.ConfigManager;
+import org.photonvision.vision.camera.CameraQuirk;
+import org.photonvision.vision.processes.VisionSourceSettables;
+
+public class GenericUSBCameraSettables extends VisionSourceSettables {
+    // We need to remember the last exposure set when exiting
+    // auto exposure mode so we can restore it
+    protected double lastExposureRaw = -1;
+
+    // Some cameras need logic where we re-apply brightness after
+    // changing exposure
+    protected int lastBrightness = -1;
+
+    protected VideoProperty exposureAbsProp = null;
+    protected VideoProperty autoExposureProp = null;
+
+    protected double minExposure = 1;
+    protected double maxExposure = 80000;
+
+    protected static final int PROP_AUTO_EXPOSURE_ENABLED = 3;
+    protected static final int PROP_AUTO_EXPOSURE_DISABLED = 1;
+
+    protected int whiteBalanceTemperature = 4000;
+
+    protected UsbCamera camera;
+    protected CameraConfiguration configuration;
+
+    public GenericUSBCameraSettables(CameraConfiguration configuration, UsbCamera camera) {
+        super(configuration);
+
+        this.configuration = configuration;
+        this.camera = camera;
+
+        getAllVideoModes();
+
+        if (!configuration.cameraQuirks.hasQuirk(CameraQuirk.StickyFPS)) {
+            if (!videoModes.isEmpty()) {
+                setVideoMode(videoModes.get(0)); // fixes double FPS set
+            }
+        }
+    }
+
+    protected void setUpExposureProperties() {
```

Comment:
Can this happen in the constructor? Just curious why this is split out.

Reply:
maybe, smol brain on java.

Here's my thought process:

The key output is to set the properties that the UI controls will modify, as well as their ranges. To do this, three possible flavors today:

1) Search among multiple valid options, and trusts the min/max reported from the driver. The Generic (superclass) implementation does this.
2) Do the generic search, but use hardcoded min/max (arducams do this)
3) Don't bother searching, trust cscore, set properties to null (because they're unused), and hardcode some generic limits (windows lifecam does this)

Reply:
I mean why not call setUpExposureProperties() at the end of the constructor?